cloudant-client/src/test/java/com/cloudant/tests/CloudFoundryServiceTest.java

```diff
@@ -83,14 +148,74 @@ public String toJson() {
+                "user", "password");
+        CloudantClient client = ClientBuilder
+                .bluemix(vcap.toJson(), serviceName, "test_bluemix_service_1")
+                .disableSSLAuthentication()
+                .build();
+        this.assertMockServerInfo(client, true);
+        // One request to _session then one to get info
+        assertEquals(3, server.getRequestCount(), "There should be three requests");
+    }
+
+    private void assertMockServerInfo(CloudantClient client, boolean isCookieAuth) throws Exception {
+        if (isCookieAuth) {
+            // Mock a 200 OK for the _session request
+            server.enqueue(MockWebServerResources.OK_COOKIE);
+        } else {
+
+
+            // Request sequence
+            // _iam_session request to get Cookie
+            // GET request -> 200 with a Set-Cookie
+            server.enqueue(OK_IAM_COOKIE);
+
+            mockIamServer.enqueue(new MockResponse().setResponseCode(200).setBody(IAM_TOKEN));
+        }
+        // 200 with the server info
+        MockResponse serverInfoResponse = new MockResponse().setResponseCode(200)
+                .setBody("{\"couchdb\":\"Welcome\"," +
+                        "\"version\":\"2.1.1\",\"vendor\":{\"name\":\"IBM Cloudant\"," +
+                        "\"version\":\"6919\",\"variant\":\"paas\"},\"features\":[\"geo\",\"scheduler\"," +
+                        "\"iam\"]}");
+        server.enqueue(serverInfoResponse);
+
+        HttpConnection conn = client.executeRequest(Http.GET(client.getBaseUri()));
+        // Consume response stream and assert ok: true
+        String responseStr = conn.responseAsString();
+        assertNotNull(responseStr);
+
+        if (isCookieAuth) {
+            // _session request
+            RecordedRequest cookieRequest = server.takeRequest();
+            assertEquals("POST", cookieRequest.getMethod(), "The request method should be POST");
+            assertEquals("/_session", cookieRequest.getPath(), "The request should be to the _session path");
+
+            // server info request
+            RecordedRequest request = server.takeRequest();
+            assertEquals("GET", request.getMethod(), "The request method should be GET");
+            assertEquals("/", request.getPath(), "The request should be for /");
+        } else {
+            MockWebServerResources.assertMockIamRequests(server, mockIamServer);
+        }
+
+        // assert CouchDB version
+        assertTrue(responseStr.contains("\"version\":\"2.1.1\""));
+
+        // assert same request details for serverVersion method
+        server.enqueue(serverInfoResponse);
+        client.serverVersion();
+
+        // server info request
+        RecordedRequest request = server.takeRequest();
+        assertEquals("GET", request.getMethod(), "The request method should be GET");
+        assertEquals("/", request.getPath(), "The request should be to the _session path");
```

Comment:
The assertion message is wrong here, but I don't think this is needed anyway, see above.